spatialprofilingtoolbox/workflow/assets/cggnn.nf

```diff
@@ -133,6 +135,10 @@ process finalize_graphs {
 process train {
     publishDir '.', mode: 'copy'
 
+    // beforeScript 'export DOCKER_IMAGE=$(if [[ "${cuda}" == "true" ]]; then echo "nadeemlab/spt-cg-gnn:cuda-latest"; else echo "nadeemlab/spt-cg-gnn:latest"; fi)'
+
+    // container "\${DOCKER_IMAGE}"
```

Comment:
We had some discussion about how we want to implement this train step here, and settled on replacing the container with a very unique string that's replaced with the cuda or non-cuda version of container depending on your inputs to `spt workflow configure`. Does that need to happen in this PR, or can that wait for later? Either when closing #272 or as a later feature enhancement, instead letting the container default to non-CUDA for now?

Reply:
I think it's part of the current issue. It means establishing the pattern whereby a docker repository name is configurable at runtime and used in this specific place. Check the Nextflow docs to see if it's possible to do process-specific docker container requests in the nextflow.config, so that we wouldn't have to alter the `.nf` definition. Otherwise use something like

```
// container ZZZ_GRAPH_PROCESSING_CONTAINER_NAME_ZZZ
```
in the `.nf` file and
```py
re.sub(special_string, container_name, nf_file_contents)`
```
during generation of the `.nf` file.